engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala

```diff
@@ -93,12 +100,22 @@ class EnsureCompiledJob(protected val files: Iterable[File])
     compile(module)
     val changeset = applyEdits(new File(module.getPath))
     compile(module)
-      .map { module =>
+      .map { moduleScope =>
         val cacheInvalidationCommands =
-          buildCacheInvalidationCommands(changeset, module.getLiteralSource)
+          buildCacheInvalidationCommands(
+            changeset,
+            moduleScope.getModule.getLiteralSource
+          )
         runInvalidationCommands(cacheInvalidationCommands)
         if (ctx.executionService.getContext.isProjectSuggestionsEnabled) {
-          analyzeModule(module, changeset)
+          if (ctx.executionService.getContext.isGlobalSuggestionsEnabled) {
+            getCompiledModules(moduleScope).foreach(analyzeModuleInScope)
+          } else {
+            val projectModules = getCompiledModules(moduleScope)
+              .filter(m => rootName(m.getName) == rootName(module.getName))
+            projectModules.foreach(analyzeModuleInScope)
+          }
+          analyzeModule(moduleScope.getModule, changeset)
```

Comment:
This logic is now really complicated and I don't see anywhere where it would be documented.

I think it would be good to add a high-level description of what this should do and maybe why. Of course no need to document the details that the code tells itself, by just by looking at the code I see that `if global suggestions are enabled then something happens` but I do not see why.

Reply:
Also, shouldn't the global suggestions thing fire regardless of project suggestions? Or is it the case that `global implies project`.